dviz_core/Goal[] goals
int32 current_goal